src/deluge/gui/ui/save/save_song_ui.cpp

```diff
@@ -352,18 +356,15 @@ bool SaveSongUI::performSave(bool mayOverwrite) {
 						if (result) {
 							D_PRINTLN("read fail");
 fail3:
-							f_close(&recorderFileSystemStuff.currentFile);
 							error = Error::UNSPECIFIED;
 							goto failAfterOpeningSourceFile;
 						}
 						if (!bytesRead) {
 							break; // Stop, on rare case where file ended right at end of last cluster
 						}
 
-						UINT bytesWritten;
-						result = f_write(&recorderFileSystemStuff.currentFile, storageManager.fileClusterBuffer,
-						                 bytesRead, &bytesWritten);
-						if (result || bytesWritten != bytesRead) {
+						auto written = created.value().write({(std::byte*)storageManager.fileClusterBuffer, bytesRead});
```

Comment:
is there an unwrap equivalent so that this can just be created.write? We've already checked that opening succeeded in this path

Reply:
actually, yeah. the dereference operators `*` and `->` haha